packages/builders/src/messages/embed/Assertions.ts

```diff
@@ -1,99 +1,53 @@
-
-export const urlPredicate = s
-	.string()
-	.url({
-		allowedProtocols: ['http:', 'https:'],
-	})
-	.nullish()
-	.setValidationEnabled(isValidationEnabled);
-
-export const embedAuthorPredicate = s
-	.object({
-		name: authorNamePredicate,
-		iconURL: imageURLPredicate,
-		url: urlPredicate,
-	})
-	.setValidationEnabled(isValidationEnabled);
-
-export const RGBPredicate = s
-	.number()
-	.int()
-	.greaterThanOrEqual(0)
-	.lessThanOrEqual(255)
-	.setValidationEnabled(isValidationEnabled);
-export const colorPredicate = s
-	.number()
-	.int()
-	.greaterThanOrEqual(0)
-	.lessThanOrEqual(0xffffff)
-	.or(s.tuple([RGBPredicate, RGBPredicate, RGBPredicate]))
-	.nullable()
-	.setValidationEnabled(isValidationEnabled);
+export const namePredicate = z.string().min(1).max(256);
 
-export const descriptionPredicate = s
+export const iconURLPredicate = z
 	.string()
-	.lengthGreaterThanOrEqual(1)
-	.lengthLessThanOrEqual(4_096)
-	.nullable()
-	.setValidationEnabled(isValidationEnabled);
+	.url()
+	.refine(refineURLPredicate(['http:', 'https:', 'attachment:']), {
+		message: 'Invalid protocol for icon URL. Must be http:, https:, or attachment:',
+	});
 
-export const footerTextPredicate = s
+export const URLPredicate = z
 	.string()
-	.lengthGreaterThanOrEqual(1)
-	.lengthLessThanOrEqual(2_048)
-	.nullable()
-	.setValidationEnabled(isValidationEnabled);
-
-export const embedFooterPredicate = s
+	.url()
+	.refine(refineURLPredicate(['http:', 'https:']), { message: 'Invalid protocol for URL. Must be http: or https:' });
+
+export const embedFieldPredicate = z.object({
+	name: namePredicate,
+	value: z.string().min(1).max(1_024),
+	inline: z.boolean().optional(),
+});
+
+export const embedAuthorPredicate = z.object({
+	name: namePredicate,
+	icon_url: iconURLPredicate.optional(),
+	url: URLPredicate.optional(),
+});
+
+export const embedFooterPredicate = z.object({
+	text: z.string().min(1).max(2_048),
+	icon_url: iconURLPredicate.optional(),
+});
+
+export const embedPredicate = z
 	.object({
-		text: footerTextPredicate,
-		iconURL: imageURLPredicate,
+		title: namePredicate.optional(),
+		description: z.string().min(1).max(4_096).optional(),
+		url: URLPredicate.optional(),
+		timestamp: z.string().optional(),
+		color: z.number().int().min(0).max(0xffffff).optional(),
+		footer: embedFooterPredicate.optional(),
+		image: z.object({ url: URLPredicate }).optional(),
+		thumbnail: z.object({ url: URLPredicate }).optional(),
+		author: embedAuthorPredicate.optional(),
+		fields: z.array(embedFieldPredicate).max(25).optional(),
 	})
-	.setValidationEnabled(isValidationEnabled);
-
-export const timestampPredicate = s.union([s.number(), s.date()]).nullable().setValidationEnabled(isValidationEnabled);
-
-export const titlePredicate = fieldNamePredicate.nullable().setValidationEnabled(isValidationEnabled);
+	.refine(
+		(value) => {
+			return Object.keys(value).length > 0;
+		},
+		{ message: 'Embed must have at least one property set' },
+	);
```

Comment:
This isn't quite right. 

One of these fields must be present:
- title
- description
- fields (at least one)
- footer.text
- author.name
- image.url
- thumbnail.url


Additionally, this is missing the 6_000 maximum character check.